crates/holochain_sqlite/src/schema/cell/initial.sql

```diff
@@ -0,0 +1,138 @@
+--       like between Update and Delete
+CREATE TABLE IF NOT EXISTS Header (
+    hash             BLOB           PRIMARY KEY ON CONFLICT IGNORE,
+    type             TEXT           NOT NULL,
+    seq              INTEGER        NOT NULL,
+    author           BLOB           NOT NULL,
+
+    blob             BLOB           NOT NULL,
+    prev_hash        BLOB           NULL,
+
+    -- Create / Update
+    entry_hash       BLOB           NULL,
+    entry_type       TEXT           NULL,  -- The opaque EntryType
+    private_entry    INTEGER        NULL,  -- BOOLEAN
+
+    -- Update
+    original_entry_hash   BLOB      NULL,
+    original_header_hash  BLOB      NULL,
+
+    -- Delete
+    deletes_entry_hash    BLOB      NULL,
+    deletes_header_hash   BLOB      NULL,
+
+    -- CreateLink
+    -- NB: basis_hash can't be foreign key, since it could map to either
+    --     Entry or Header
+    -- FIXME: @freesig Actually this can only be an EntryHash.
+    -- Links can't be on headers.
+    base_hash        BLOB           NULL,
+    zome_id          INTEGER        NULL,
+    tag              BLOB           NULL,
+
+    -- DeleteLink
+    create_link_hash    BLOB           NULL,
+
+    -- AgentValidationPkg
+    membrane_proof   BLOB           NULL,
+
+    -- OpenChain / CloseChain
+    prev_dna_hash    BLOB           NULL
+
+    -- We can't have any of these constraint because 
+    -- the element authority doesn't get the create link for a remove link. @freesig
+    -- FOREIGN KEY(entry_hash) REFERENCES Entry(hash)
+    -- FOREIGN KEY(original_entry_hash) REFERENCES Entry(hash),
+    -- FOREIGN KEY(original_header_hash) REFERENCES Header(hash),
+    -- FOREIGN KEY(deletes_entry_hash) REFERENCES Entry(hash)
+    -- FOREIGN KEY(deletes_header_hash) REFERENCES Header(hash),
+    -- FOREIGN KEY(create_link_hash) REFERENCES Header(hash)
+);
+CREATE INDEX IF NOT EXISTS Header_type_idx ON Header ( type );
+CREATE INDEX IF NOT EXISTS Header_author ON Header ( author );
+
+
+-- NB: basis_hash, header_hash, and entry_hash, in general, will have
+--     duplication of data. Could rethink these a bit.
+CREATE TABLE IF NOT EXISTS DhtOp (
+    hash             BLOB           PRIMARY KEY ON CONFLICT IGNORE,
+    type             TEXT           NOT NULL,
+    basis_hash       BLOB           NOT NULL,
+    header_hash      BLOB           NOT NULL,
+    is_authored      INTEGER        NOT NULL,      -- BOOLEAN
+    require_receipt  INTEGER        NOT NULL,      -- BOOLEAN
+
+    -- This is the order that process ops should result 
+    -- in dependencies before dependants.
+    -- See OpOrder.
+    op_order        TEXT           NOT NULL,
+
+    -- If this is null then validation is still in progress.
+    validation_status INTEGER       NULL,
+
+    when_integrated  INTEGER NULL,          -- DATETIME
+    -- We need nanosecond accuracy which doesn't fit in 
+    -- an INTEGER.
+    when_integrated_ns  BLOB NULL,          -- DATETIME
```

Comment:
is there a reason we need the less accurate one if we have the more accurate one?

Reply:
Yeh to compare with timestamps because we can't do `DhtOp.when_integrated >= :from` with a BLOB